### PR TITLE
Start with phpstan

### PR DESCRIPTION
**Pull request by szepeviktor**
@dsnopek What do you think about doing this business with PHPStan?

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        - %currentWorkingDirectory%/modules/blog/blog.module
+        - %currentWorkingDirectory%/modules/blog/blog.pages.inc
+        - %currentWorkingDirectory%/modules/filter/filter.pages.inc
+        - %currentWorkingDirectory%/modules/filter/filter.admin.inc
+        - %currentWorkingDirectory%/modules/filter/filter.module
+        - %currentWorkingDirectory%/modules/filter/filter.install
+        - %currentWorkingDirectory%/modules/help/help.admin.inc
+        - %currentWorkingDirectory%/modules/help/help.module
+#    autoload_directories:
+    paths:
```
Comment (szepeviktor) [inline phpstan.neon:147]: This checks _includes_ automatically

**Comment by dsnopek** [comment]
Thanks!

However, these sort of mass changes based on automatic tools make me a little nervous. Paging through the diff, it definitely seems like this is catching some possible issues that need to be fixed, but it's easier to address individual changes that come from actual testing (and that makes them easier to test too). It could take a while to sift through all of this!

**Comment by szepeviktor** [comment]
Thank you for your response.

That is why I'm asking.
PHPStan is a great source of profit, on Level 7 your PHP code may be much better than before.
How to proceed?

**Comment by dsnopek** [comment]
Well, each of these change is going to need to be evaluated and tested. The documentation changes are sure to be fine. I guess I prefer changing fewer things, even if the change is "more correct" just in case the change has unexpected consequences - up until now we've only been changing this that verifiably cause issues.

**Comment by szepeviktor** [comment]
@dsnopek OK. I consider this an answer and stop sending fixes PHPStan-discovered issues.

**Comment by szepeviktor** [comment]
> verifiably cause issues

@dsnopek Do you think PHPStan reports false positives?

If so please close this PR.
If not could you please merge it?

**Comment by dsnopek** [comment]
Unfortunately, I don't know. Someone will need to go through these one by one, examine the code and then test it. I haven't had time to examine this in that depth.

**Comment by szepeviktor** [comment]
I see.

File: includes/batch.inc [hunk 2]
```diff
@@ -143,7 +143,7 @@ function _batch_progress_page_nojs() {
     print $fallback;
 
     // Perform actual processing.
-    list($percentage, $message) = _batch_process($batch);
+    list($percentage, $message) = _batch_process();
```
Comment (rmuit) [inline includes/batch.inc:146]: Just a note for whoever happens to come across it: this is indeed a proper change. $batch was never used as an argument to _batch_process(), at least not in D6's lifetime. _batch_process() gets the batch not from the function argument but from its first line
```
$batch =& batch_get();
```

**Comment by dsnopek** [comment]
I'm going to close this. I appreciate that there is probably some good stuff in here, but it's just more changes than we have the bandwidth to evaluate and test, given how risky it is to make big changes to a legacy code base.

**Comment by szepeviktor** [comment]
All right.
I understand your position on the safe side.

**Pull request closed without merging**